### PR TITLE
Remove extraneous whitespace - #26275

### PR DESCRIPTION
**Pull request by DanielRuf**
### Description (*)

This PR removes the extraneous spaces from the labels.

### Fixed Issues (if relevant)

1. magento/magento2#26275

### Manual testing scenarios (*)

1. Go to checkout
2. There should not be any extra whitespace before the asterisk / required icon.

Without this PR:
grafik

With the fixes in the PR:
grafik

### Contribution checklist (*)
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Comment by DanielRuf** [comment]
> @magento-automated-testing Static Tests Failing after 18m — Build reports

Huh, we can not see any details what and why it fails?

**Comment by DanielRuf** [comment]
Which violations are meant here?

https://public-results-storage-prod.magento-testing-service.engineering/reports/magento/magento2/pull/26285/d17f81ec84bfc2dbd18f5b8133fd880c/Statics/allure-report-ce/index.html#categories/78ae151bd5d3aa3d8f50dc36ec4ec082/abecdf50b283f5ae/

```
PHP Code Sniffer detected 2 violation(s): 

FILE: ...code/Magento/Customer/view/frontend/templates/address/edit.phtml
----------------------------------------------------------------------
FOUND 0 ERRORS AND 34 WARNINGS AFFECTING 24 LINES
----------------------------------------------------------------------
  25 | WARNING | [x] Expected 0 space(s) after closing parenthesis;
     |         |     found 1
  29 | WARNING | [x] Expected 0 space(s) after closing parenthesis;
     |         |     found 1
  33 | WARNING | [x] Expected 0 space(s) after closing parenthesis;
     |         |     found 1
  40 | WARNING | [ ] The use of helpers in templates is discouraged.
     |         |     Use ViewModel instead.
  40 | WARNING | [ ] Line exceeds 120 characters; contains 137
     |         |     characters
  42 | WARNING | [ ] Line exceeds 120 characters; contains 146
     |         |     characters
  52 | WARNING | [ ] The use of helpers in templates is discouraged.
     |         |     Use ViewModel instead.
  52 | WARNING | [x] Expected 0 space(s) after closing parenthesis;
     |         |     found 1
  52 | WARNING | [ ] Line exceeds 120 characters; contains 142
     |         |     characters
  70 | WARNING | [ ] The use of helpers in templates is discouraged.
     |         |     Use ViewModel instead.
  70 | WARNING | [x] Expected 0 space(s) after closing parenthesis;
     |         |     found 1
  80 | WARNING | [ ] The use of helpers in templates is discouraged.
     |         |     Use ViewModel instead.
  80 | WARNING | [ ] Line exceeds 120 characters; contains 170
     |         |     characters
  86 | WARNING | [ ] Line exceeds 120 characters; contains 140
     |         |     characters
  92 | WARNING | [ ] The use of helpers in templates is discouraged.
     |         |     Use ViewModel instead.
  92 | WARNING | [ ] Line exceeds 120 characters; contains 164
     |         |     characters
 103 | WARNING | [ ] Line exceeds 120 characters; contains 160
     |         |     characters
 104 | WARNING | [ ] Line exceeds 120 characters; contains 121
     |         |     characters
 111 | WARNING | [ ] The use of helpers in templates is discouraged.
     |         |     Use ViewModel instead.
 111 | WARNING | [ ] Line exceeds 120 characters; contains 279
     |         |     characters
 124 | WARNING | [ ] The use of helpers in templates is discouraged.
     |         |     Use ViewModel instead.
 124 | WARNING | [ ] Line exceeds 120 characters; contains 196
     |         |     characters
 131 | WARNING | [ ] Line exceeds 120 characters; contains 149
     |         |     characters
 137 | WARNING | [x] Expected 0 space(s) after closing parenthesis;
     |         |     found 1
 141 | WARNING | [x] Expected 0 space(s) after closing parenthesis;
     |         |     found 1
 148 | WARNING | [x] Expected 0 space(s) after ELSE keyword; 1 found
 152 | WARNING | [x] Expected 0 space(s) after closing parenthesis;
     |         |     found 1
 156 | WARNING | [x] Expected 0 space(s) after closing parenthesis;
     |         |     found 1
 163 | WARNING | [x] Expected 0 space(s) after ELSE keyword; 1 found
 192 | WARNING | [ ] Line exceeds 120 characters; contains 131
     |         |     characters
 197 | WARNING | [ ] The use of helpers in templates is discouraged.
     |         |     Use ViewModel instead.
 197 | WARNING | [ ] Line exceeds 120 characters; contains 123
     |         |     characters
 199 | WARNING | [ ] The use of helpers in templates is discouraged.
     |         |     Use ViewModel instead.
 199 | WARNING | [ ] Line exceeds 120 characters; contains 154
     |         |     characters
----------------------------------------------------------------------
PHPCBF CAN FIX THE 11 MARKED SNIFF VIOLATIONS AUTOMATICALLY
----------------------------------------------------------------------



Failed asserting that 2 matches expected 0.
```

**Comment by marcoaacoliveira** [comment]
@DanielRuf Thanks for your contribution. You changed a file and when you do it Magento will run static tests against this file, so even if the problem is older than your changes you must fix all it to get pass in the tests.

Specifically in your case changes aren't hard... 

**You must do 2 types of refactoring:**

### First:
You must change all `$this->helper` references for viewModel approach, which is basically:

- Instead of using $this->helper, you must create a new block viewModel (if it doesn't exists), and it must be the responsible for all data or functionality (handling data matters). In [DevDocs](https://devdocs.magento.com/guides/v2.3/extension-dev-guide/view-models.html) you can find how to do it step by step.

### Second:
The easiest one, you just have to guarantee that PHP code lines aren't too big, it must be equal or lesser than 120 characters per line.

**Comment by DanielRuf** [comment]
> @DanielRuf Thanks for your contribution. You changed a file and when you do it Magento will run static tests against this file, so even if the problem is older than your changes you must fix all it to get pass in the tests.

I am very aware of that.

> The easiest one, you just have to guarantee that PHP code lines aren't too big, it must be equal or lesser than 120 characters per line.

Which would produce the same issue again.



> You must change all `$this->helper` references for viewModel approach, which is basically:
> 
>     * Instead of using $this->helper, you must create a new block viewModel (if it doesn't exists), and it must be the responsible for all data or functionality (handling data matters). In [DevDocs](https://devdocs.magento.com/guides/v2.3/extension-dev-guide/view-models.html) you can find how to do it step by step.

I've already seen that in the report. I don't really know here as I was mostly doing frontend stuff in Magento 2 projects. I'm not so deep in these helper stuff.

**Comment by DanielRuf** [comment]
> @DanielRuf Thanks for your contribution. You changed a file and when you do it Magento will run static tests against this file, so even if the problem is older than your changes you must fix all it to get pass in the tests.

I am very aware of that.

> The easiest one, you just have to guarantee that PHP code lines aren't too big, it must be equal or lesser than 120 characters per line.

Which would produce the same issue again.



> You must change all `$this->helper` references for viewModel approach, which is basically:
> 
>     * Instead of using $this->helper, you must create a new block viewModel (if it doesn't exists), and it must be the responsible for all data or functionality (handling data matters). In [DevDocs](https://devdocs.magento.com/guides/v2.3/extension-dev-guide/view-models.html) you can find how to do it step by step.

I've already seen that in the report. I don't really know here as I was mostly doing frontend stuff in Magento 2 projects. I do not know much about these helpers.

**Comment by DanielRuf** [comment]
> @DanielRuf Thanks for your contribution. You changed a file and when you do it Magento will run static tests against this file, so even if the problem is older than your changes you must fix all it to get pass in the tests.

I am very aware of that.

> The easiest one, you just have to guarantee that PHP code lines aren't too big, it must be equal or lesser than 120 characters per line.

Which would produce the same issue again.



> You must change all `$this->helper` references for viewModel approach, which is basically:
> 
> * Instead of using $this->helper, you must create a new block viewModel (if it doesn't exists), and it must be the responsible for all data or functionality (handling data matters). In [DevDocs](https://devdocs.magento.com/guides/v2.3/extension-dev-guide/view-models.html) you can find how to do it step by step.

I've already seen that in the report. I don't really know here as I was mostly doing frontend stuff in Magento 2 projects. I do not know much about these helpers.

**Comment by DanielRuf** [comment]
Well it says "2 violations" which is not very clear. Also warnings should be actual errors then.

**Comment by marcoaacoliveira** [comment]
Indeed, maybe a maintainer can be more helpful to this matter.

**Comment by DanielRuf** [comment]
Currently trying to fix the outstanding PHPCS violations.

**Comment by marcoaacoliveira** [comment]
Good luck and have fun :)

**Comment by DanielRuf** [comment]
Hm, I see no real step for step description for migrating the `\Magento\Customer\Helper\Address` to a ViewModel. Also the link in the DevDocs at the bottom does not seem to point to the mentioned `view_model` part.

**Comment by DanielRuf** [comment]
Not sure about the helpers as I have mostly only worked on the frontend part of Magento 2.

Would be great to get some help with this.

**Comment by DanielRuf** [comment]
I will check the failing tests later.

File: app/code/Magento/Customer/view/frontend/templates/address/edit.phtml [hunk 1]
```diff
@@ -5,10 +5,27 @@
...
+<?php $_region = $block->getAttributeData()->getFrontendLabel('region'); ?>
+<?php $_selectRegion = 'Please select a region, state or province.'; ?>
+<?php $_displayAll = $block->getConfig('general/region/display_all'); ?>
+
+<?php $_vatidValidationClass = $viewModel->addressGetAttributeValidationClass('vat_id'); ?>
+<?php $_cityValidationClass = $viewModel->addressGetAttributeValidationClass('city'); ?>
+<?php $_postcodeValidationClass_value = $viewModel->addressGetAttributeValidationClass('postcode'); ?>
+<?php $_postcodeValidationClass = $_postcodeValidationClass_value; ?>
+<?php $_streetValidationClass = $viewModel->addressGetAttributeValidationClass('street'); ?>
+<?php $_streetValidationClass = trim(str_replace('required-entry', '', $_streetValidationClass)); ?>
```
Comment (nuzil) [inline app/code/Magento/Customer/view/frontend/templates/address/edit.phtml]: Looks like here removed validation for street as required field globally for all streets. In initial implementation it was so only for streets>1

**Review by nuzil — CHANGES_REQUESTED** [review]

File: app/code/Magento/Customer/view/frontend/templates/address/edit.phtml (hunk 1 above)
Comment (DanielRuf) [reply app/code/Magento/Customer/view/frontend/templates/address/edit.phtml]: Not sure what you exactly mean. Can you explain in detail? I copied the lines and changed just a few variable names.

Comment (DanielRuf) [reply app/code/Magento/Customer/view/frontend/templates/address/edit.phtml]: Sigh, such code and the 120 linelength rule are not great. I have pushed a change which should work.

Thanks for catching this.

**Comment by nuzil** [comment]
Looks like you got the point :-)

**Review by nuzil — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @nuzil, thank you for the review.
[ENGCOM-6581](https://jira.corp.magento.com/browse/ENGCOM-6581) has been created to process this Pull Request
:eight_spoked_asterisk: @nuzil, could you please add one of the following labels to the Pull Request?

| Label | Description |
|--------------------------------|-----------------------------------------------------------------|
| ```Auto-Tests: Covered``` | All changes in Pull Request is covered by auto-tests |
| ```Auto-Tests: Not Covered``` | Changes in Pull Request requires coverage by auto-tests |
| ```Auto-Tests: Not Required``` | Changes in Pull Request does not require coverage by auto-tests |

File: app/code/Magento/Customer/ViewModel/Address.php [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+namespace Magento\Customer\ViewModel;
+
+use Magento\Directory\Helper\Data as DataHelper;
+use Magento\Customer\Helper\Address as AddressHelper;
+use Magento\Framework\Exception\NoSuchEntityException;
+use Magento\Framework\View\Element\Block\ArgumentInterface;
+
+/**
+ * Customer address view model.
+ */
```
Comment (Nazar65) [inline app/code/Magento/Customer/ViewModel/Address.php]: Looks like There is two Description for a class

File: app/code/Magento/Customer/ViewModel/Address.php [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+  * Address view model
+  */
+class Address implements ArgumentInterface
+{
+    /**
+     * Data helper
+     *
+     * @var DataHelper
+     */
+    private $helperData;
```
Comment (Nazar65) [inline app/code/Magento/Customer/ViewModel/Address.php:22]:
```
   /**
     * @var DataHelper $helperData
     */
    private $helperData;
```

File: app/code/Magento/Customer/ViewModel/Address.php [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+
+    /**
+     * Address helper
+     *
+     * @var AddressHelper
```
Comment (Nazar65) [inline app/code/Magento/Customer/ViewModel/Address.php:25]: /**
     * @var AddressHelper $helperAddress
     */
    private $helperAddress;

File: app/code/Magento/Customer/ViewModel/Address.php [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+        $this->helperData= $helperData;
+        $this->helperAddress= $helperAddress;
+    }
+
+    /**
+     * Get string with frontend validation classes for attribute
+     *
+     * @param string $attributeCode
+     *
+     * @return string
```
Comment (Nazar65) [inline app/code/Magento/Customer/ViewModel/Address.php:48]: remove blank line

File: app/code/Magento/Customer/ViewModel/Address.php [hunk 6]
```diff
@@ -0,0 +1,125 @@
...
+     */
+    public function __construct(
+        DataHelper $helperData,
+        AddressHelper $helperAddress
+    ) {
+        $this->helperData= $helperData;
+        $this->helperAddress= $helperAddress;
+    }
+
+    /**
```
Comment (Nazar65) [inline app/code/Magento/Customer/ViewModel/Address.php:44]:
```
    /**
     * Get string with frontend validation classes for attribute
     *
     * @param string $attributeCode
     * @return string
     * @throws \Magento\Framework\Exception\LocalizedException
     */
```

File: app/code/Magento/Customer/ViewModel/Address.php [hunk 7]
```diff
@@ -0,0 +1,125 @@
...
+     * @return string
+     *
+     * @throws \Magento\Framework\Exception\LocalizedException
+     */
+    public function dataGetAttributeValidationClass($attributeCode)
+    {
+        return $this->helperData->getAttributeValidationClass($attributeCode);
+    }
+
+    /**
```
Comment (Nazar65) [inline app/code/Magento/Customer/ViewModel/Address.php:56]:
```
   /**
     * Get string with frontend validation classes for attribute
     *
     * @param string $attributeCode
     * @return string
     * @throws \Magento\Framework\Exception\LocalizedException
     */
```
```

**Review by Nazar65 — CHANGES_REQUESTED** [review]

File: app/code/Magento/Customer/ViewModel/Address.php (hunk 3 above)
Comment (nuzil) [reply app/code/Magento/Customer/ViewModel/Address.php:22]: @Nazar65 
For the class properties you dont need to duplicate variable name.
@var DataHelper is correct. Or there are some Guidelines about that?

File: app/code/Magento/Customer/ViewModel/Address.php (hunk 4 above)
Comment (nuzil) [reply app/code/Magento/Customer/ViewModel/Address.php:25]: For the class properties you dont need to duplicate variable name.
@var DataHelper is correct. Or there are some Guidelines about that?

File: app/code/Magento/Customer/ViewModel/Address.php (hunk 5 above)
Comment (DanielRuf) [reply app/code/Magento/Customer/ViewModel/Address.php:48]: Afaik the linting / PHPCS had a problem with the missing empty line. Check the failing checks from the previous CI builds.

File: app/code/Magento/Customer/ViewModel/Address.php (hunk 3 above)
Comment (Nazar65) [reply app/code/Magento/Customer/ViewModel/Address.php:22]: @nuzil yes you are right , @var DataHelper, will be enough

**Review by Nazar65 — APPROVED** [review]

**Comment by engcom-Delta** [comment]
:heavy_check_mark: QA passed

**Comment by DanielRuf** [comment]
Static Tests fail now. I will revert a few of the proposed changes.

https://github.com/magento/magento2/runs/391004749

**Comment by engcom-Golf** [comment]
@DanielRuf i will take care of satic test

**Comment by DanielRuf** [comment]
> @DanielRuf i will take care of satic test

I already did. No need to.

**Pull request merged**